### ISSUE TITLE
Issues with disabled reduction during gradient accumulation

### ISSUE THREAD

**Issue by krishansubudhi** [NVIDIA/apex#708]
I want to disable all-reduce during gradient accumulation. If my gradient accumulation is 2, I want to enable all reduce every other step. This will speed up my training. 

Using this technique with Apex results in out of sync master gradients and the model does not converge well.

Detailed blog:
https://krishansubudhi.github.io/deeplearning/2020/02/06/apex-gradient-accumulation.html

![image](https://user-images.githubusercontent.com/11926616/74045049-f697b380-4980-11ea-9b91-d7feef44eb7b.png)

**Comment by krishansubudhi** [comment]
Found a solution to this issue. Details in this blog .
https://krishansubudhi.github.io/deeplearning/2020/02/06/apex-gradient-accumulation.html
